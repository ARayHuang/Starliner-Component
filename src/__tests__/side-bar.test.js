import React from 'react';
import { shallow, mount } from 'enzyme';
import SideBar from '../components/side-bar';

describe('side bar ', () => {
	it('handle default props', () => {
		const  {
			isOpen,
			isFixedWidth,
			width,
			collapseWidth,
			onMouseEnter,
			onMouseLeave
		} = SideBar.defaultProps;

		expect(isOpen).toEqual(false);
		expect(isFixedWidth).toEqual(false);
		expect(width).toEqual('250px');
		expect(collapseWidth).toEqual('30px');
		expect(onMouseEnter).toBeInstanceOf(Function);
		expect(onMouseLeave).toBeInstanceOf(Function);
	});

	it('should render correctly', () => {
		const props = {
			isOpen: true,
			width: '150px',
			collapseWidth: '40px',
			styled: {
				color: '#FFF',
				bg: '#0c1d47'
			},
			children: 'mock-children',
			collapseChildren: 'mock-collapseChildren',
			onMouseEnter: () => {},
			onMouseLeave: () => {}
		}

		const wrapper = shallow(
			<SideBar {...props}/>
		)

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const props = {
			isOpen: true,
			width: '150px',
			collapseWidth: '40px',
			styled: {
				color: '#FFF',
				bg: '#0c1d47'
			},
			children: 'mock-children',
			collapseChildren: 'mock-collapseChildren',
			onMouseEnter: () => {},
			onMouseLeave: () => {}
		}
		const wrapper = mount(
			<SideBar {...props}/>
		);

		expect(wrapper.props().isOpen).toEqual(props.isOpen);
		expect(wrapper.props().width).toEqual(props.width);
		expect(wrapper.props().collapseWidth).toEqual(props.collapseWidth);
		expect(wrapper.props().styled).toEqual(props.styled);
		expect(wrapper.props().children).toEqual(props.children);
		expect(wrapper.props().collapseChildren).toEqual(props.collapseChildren);
		expect(wrapper.props().onMouseEnter).toEqual(props.onMouseEnter);
		expect(wrapper.props().onMouseLeave).toEqual(props.onMouseLeave);
	})
})
