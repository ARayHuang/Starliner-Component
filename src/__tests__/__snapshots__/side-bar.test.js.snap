// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`side bar  should render correctly 1`] = `
<Box
  as="nav"
  bg="#0c1d47"
  color="#FFF"
  display="block"
  height="100vh"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  px="3"
  py="5"
  right="0"
  transition="all 0.2s"
  width="150px"
/>
`;
